File: Warp_Core1/Warp_Core1/Warp_Image.cpp
#include "Warp.h"

template <T> WarpImage<T>::WarpImage(data_t dataType, int width, int height)
{
	this.data_type = dataType;
	this.width = width;
	this.height = height;
	data = new T[width*height];
}

File: warp5/warp5/WarpImage.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace warp5
{
    class WarpImage<T>
    {
        private Coord RA;       // object RA (Right Ascention)
        private Coord DEC;      //object Dec (declination)
        private uint width;     //image width
        private uint height;    //image height
        private string oName;   //object name
        private string notes;   //fits notes
        private T[][] data;     //image data
    }
}
